mscore/events.cpp

```diff
@@ -774,6 +774,9 @@ void ScoreView::keyPressEvent(QKeyEvent* ev)
       editData.modifiers = ev->modifiers();
       editData.s         = ev->text();
 
+      if((editData.modifiers & Qt::ControlModifier) && editData.key == Qt::Key_A) // ctrl + a (select all)
+            QTimer::singleShot(200, this, SLOT(repaint())); // repaint the ScoreView after 200ms, used to display the selection without needing to release the buttons
```

Comment:
I would say that this solution has several drawbacks which make this solution overall be not very clean, as you have mentioned in the pull request description.
- It assumes that exactly and only Ctrl+A combination can be used to select the entire text. This is currently hardcoded in text editing code anyway, but with this solution it is now hardcoded in two places, and if we ever decide to change this (for example, if we decide to allow customization of that shortcut), we will need to somehow remember to change it here as well.
- It updates the view after some delay. It is not clear why exactly this delay value is chosen, and why there should be a delay at all.
- It forces an immediate [repaint](https://doc.qt.io/qt-5/qwidget.html#repaint) of the entire view after that 200 ms interval. This means that if user presses Ctrl+A several times in a row quickly enough there is a good chance to get a noticeable lagging because of multiple repaintings of the entire view caused by these timers triggered by the keystrokes.
- Most importantly of all that, it does not fix the real problem (at least this "select all" action on text editing does not trigger updating the relevant part of view, while most of other key presses do) and just cures the symptoms of the reported issue by the code handling this exact case (Ctrl+A key press) in a special way. It is better to avoid such solutions if possible, and in this case there definitely exist a cleaner solution.

Therefore I believe this change in its current state shouldn't be merged, and a better solution is needed.
Actually while writing this I found a better solution for this but if you want to look into this issue yourself I might leave this issue to you. As a small hint, you might want to look how it works in similar cases which do not have this issue (for example, selecting text with <kbd>Shift</kbd>+arrows, <kbd>Shift</kbd>+<kbd>Home</kbd>, <kbd>Shift</kbd>+<kbd>End</kbd> keys as well as moving cursor with arrow and <kbd>Home</kbd>/<kbd>End</kbd> keys does update the view correctly without the need to release those keys).

Reply:
Thanks for the tip :^). I will look into it and return with a better solution.